types/chai/index.d.ts

```diff
@@ -523,6 +528,16 @@ declare global {
              */
             deepStrictEqual<T>(actual: T, expected: T, message?: string): void;
 
+            /**
+             * Partially matches actual and expected.
+             *
+             * T   Type of the objects.
+             * @param actual   Actual value.
+             * @param expected   Potential expected value.
+             * @param message   Message to display on error.
+             */
+            containSubset<T>(val: T, exp: Partial<T>, msg?: string): void;
```

Comment:
Double checking this, but this seems to only work for objects and a single depth.

The implementation (https://github.com/chaijs/chai/blob/ce4c403d0f60818e1f5e1d4aa472e2a2b0a1ad8a/lib/chai/core/assertions.js#L4076) appears to actually support arrays, and the whole thing is deeply matched. That and Dates are _not_ deeply matched.

This makes me wonder if this needs some sort of DeepPartial or something to capture the full behavior (and `chai-subset` itself took `any` instead of specific types).

Reply:
yeah, I was hoping to improve on the original typing but `any` might be correct here, since there is nothing really stopping anyone from passing an unrelated object?

Reply:
Probably not, no...

Reply:
I simplified it to use `any`